proj/hps_accel/gateware/stream/__init__.py

```diff
@@ -46,16 +46,26 @@
 
 This implementation was heavily influenced by the discussion at
 https://github.com/nmigen/nmigen/issues/317, and especially the existing design
-of LiteX streams.
+of LiteX streams. It relies heavily on the nmigen Record class for payload
+manipulation.
+
+Much of the power of LiteX streams comes from the ability to programatically
+manipulate stream processing code. Because that metaprogramming code is often
+difficult to write and read, this streams implementation makes a minor
+tradeoffs in favor of simplicity of metaprogramming over "everyday" stream
+use. In particular, the stream has a "payload" attribute to explicitly contain
```

Comment:
Good to know why this difference is intentional

Reply:
Mistake on my part: LiteX does have a separate payload, and then also exposes attributes directly on the Source/Sink via getattr() tricks.